public/app/core/utils/datemath.ts

```diff
@@ -3,7 +3,20 @@ import moment from 'moment';
 
 const units = ['y', 'M', 'w', 'd', 'h', 'm', 's'];
 
-export function parse(text, roundUp?, timezone?) {
+type Timezone = 'utc';
+
+/**
+ * Parses different types input to a moment instance. There is a specific formatting language that can be used
+ * if text arg is string. See unit tests for examples.
+ * @param text
+ * @param roundUp See parseDateMath function.
+ * @param timezone Only string 'utc' is acceptable here, for anything else, local timezone is used.
+ */
+export function parse(
+  text: string | moment.Moment | Date,
+  roundUp?: boolean,
+  timezone?: Timezone
+): moment.Moment | undefined {
```

Comment:
I just now realised that was never working with TS without strictNullChecks and we do not have that right now. So this should generate a lots of issues as we do not check that much for whether parse could return undefined, but as we do not have strictNullChecks it does not complain yet. If we enable strictNullChecks this will start to complain about that.

Reply:
after https://github.com/grafana/grafana/issues/14714 is done, see progress here: https://play.grafana.org/d/VEy5UXjmk/grafana-ci-code-stats?orgId=1 

I want to start add structNullChecks as well :)